design/design.md

```diff
@@ -0,0 +1,75 @@
+# Design for Z HMC OS Message Forwarder
+
+## Functionality
+
+* Runs as a process that gets OS messages from certain partitions and sends them
+  to a remote syslog service.
+
+* One forwarder process connects to one HMC and handles all of its managed CPCs.
+
+* The CPCs can be in classic mode (then the messages are from OSs in LPARs) or
+  in DPM mode (then the messages are from OSs in partitions).
+
+* The LPARs and partitions are selected by CPC name and LPAR/partition name,
+  whereby these names can be specified as a list whose items are regular
+  expressions.
+
+* On CPCs in DPM mode, newly created partitions that match the specified names
+  are automatically started to be forwarded.
+
+* On CPCs in DPM mode, partitions that have been forwarded and that are being
+  deleted, are automatically stopped to be forwarded.
+
+* TBD: New CPCs and CPCs that go away are also automatically handled.
+
+* HMC reboots are automatically recovered.
+
+* A forwarder config change requires the forwarder to be restarted.
+
+* Restarts of the forwarder process automatically detect the last OS message
+  from each LPAR/partition and resume the forwarding at the right point,
+  so that there are no duplicates and no gaps in what is sent to the syslog
+  service.
+
+## General approach
+
+The forwarder will read a config file which provides access data for the HMC,
+and a definition of CPCs and LPARs/Partitions for which the forwarding should
+be done.
+
+It retrieves the zhmcclient.LPAR/Partition objects for the matching partitions
+from the matching CPCs, and opens an OS message channel for each of them.
```

Comment:
My expectation is that you will get a 409.332 for non-active partitions. See the Open OS Message Channel operation of the partition object in the WSAPI book:

> Status code 409 (Conflict) is returned when the message interface for the operating system running in the targeted partition is not available. Some examples are when the Partition is not active, there is no operating system running in the partition, or when the operating system is not enabled for console integration.